workflow/validate/validate.go

```diff
@@ -1204,8 +1204,8 @@ func (ctx *templateValidationCtx) validateDAG(scope map[string]interface{}, tmpl
 	// Verify dependencies for all tasks can be resolved as well as template names
 	for _, task := range tmpl.DAG.Tasks {
 
-		if usingDepends && '0' <= task.Name[0] && task.Name[0] <= '9' {
-			return errors.Errorf(errors.CodeBadRequest, "templates.%s.tasks.%s name cannot begin with a digit when using 'depends'", tmpl.Name, task.Name)
+		if (usingDepends || len(task.Dependencies) > 0) && '0' <= task.Name[0] && task.Name[0] <= '9' {
```

Comment:
```suggestion
		if '0' <= task.Name[0] && task.Name[0] <= '9' {
```

Wouldn't this be better?